__tests__/Ship.test.js

```diff
@@ -0,0 +1,74 @@
+/*global describe it expect */
+const Ship = require('../src/Ship');
+const Port = require('../src/Port');
+const Itinerary = require('../src/Itinerary');
+
+describe('Ship', () => {
+    describe('with ports and an itinerary', () => {
+        let ship;
+        let rotterdam;
+        let hull;
+        let itinerary;
+
+        beforeEach(() => {
+            rotterdam = {
+                addShip : jest.fn(),
+                removeShip : jest.fn(),
+                name : 'Rotterdam' ,
+                ships : []
+            };
+
+            hull = {
+                addShip : jest.fn(),
+                removeShip : jest.fn(),
+                name : 'Hull' ,
+                ships : []
+            };
+
+            itinerary = {
+                ports: [rotterdam, hull]
+            };
+
+            ship = new Ship(itinerary);
+
+        });
+
+        it('can be instantiated' , () => {
+            expect(ship).toBeInstanceOf(Object);
+        });
+
+        it('has a starting port', () => {
+            expect(ship.currentPort).toBe(rotterdam);
+
+        });
+
+        it('can set sail', () => {
+            ship.setSail();
+
+            expect(ship.currentPort).toBeFalsy();
+            expect(rotterdam.removeShip).toHaveBeenCalledWith(ship);
+
+        });
+
+        it('gets added to port on instantiation' , () => {
+            expect(rotterdam.addShip).toHaveBeenCalledWith(ship);
+
+        });
+
+        it('can dock at a different port', () => {
+            ship.setSail();
+            ship.dock();
+
+            expect(ship.currentPort).toBe(hull);
+            expect(hull.addShip).toHaveBeenCalledWith(ship);
+        });
+
+        it('can\'t sail futher than its itinerary', () => {
+            ship.setSail();
+            ship.dock();
+
+            expect(() => ship.setSail()).toThrowError('End of itinerary reached');
+
+        });
```

Comment:
You can remove a lot of these extra blank lines here. The point of clear lines is separate different parts of the code so its easy to see the different sections at a glance, so typically you would use them between each test, before the assertions start, and between each describe block. You dont need to use then after the assertions and before the closing bracket of the function, it just adds extras spaces where its not needed

```
        it('can be instantiated' , () => {
            expect(ship).toBeInstanceOf(Object);
        });

        it('has a starting port', () => {
            expect(ship.currentPort).toBe(rotterdam);
        });

        it('can set sail', () => {
            ship.setSail();

            expect(ship.currentPort).toBeFalsy();
            expect(rotterdam.removeShip).toHaveBeenCalledWith(ship);
        });

        it('gets added to port on instantiation' , () => {
            expect(rotterdam.addShip).toHaveBeenCalledWith(ship);
        });

        it('can dock at a different port', () => {
            ship.setSail();
            ship.dock();

            expect(ship.currentPort).toBe(hull);
            expect(hull.addShip).toHaveBeenCalledWith(ship);
        });

        it('can\'t sail futher than its itinerary', () => {
            ship.setSail();
            ship.dock();

            expect(() => ship.setSail()).toThrowError('End of itinerary reached');
        });
```